### PR TITLE
Feature/cdfs 35 edit service period product

### PR DESCRIPTION
**Pull request by NitishAroraKPMG**
## Description

Edit the services for the period and flat fare ticket

## Testing instructions

Goto product details page and edit the services. Services will be edited and updated in product details

### CODE REVIEW

File: fdbt-aws/sql/n.products.sql [hunk 1]
```diff
@@ -13,3 +13,6 @@ CREATE TABLE products(
     INDEX idx_nocCode (nocCode),
     PRIMARY KEY (`id`)
 ) ENGINE = InnoDB CHARACTER SET = utf8;
+
+ALTER TABLE products 
+    ADD fareTriangleModified DATETIME DEFAULT NULL;
```
Comment (RobCattonKPMG) [inline fdbt-aws/sql/n.products.sql:18]: Could we change this so its part of the create and not an alter?

File: repos/fdbt-site/src/pages/serviceList.tsx [hunk 2]
```diff
@@ -201,6 +231,7 @@ export const getServerSideProps = async (ctx: NextPageContextWithSession): Promi
             dataSourceAttribute,
             csrfToken,
             additional,
+            editMode,
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/serviceList.tsx:234]: Same here

File: repos/fdbt-site/tests/pages/api/serviceList.test.ts [hunk 3]
```diff
@@ -117,10 +127,36 @@ describe('serviceList', () => {
...
     });
+
+    it('should update the service list when in edit mode and redirect back to products/productDetails', async () => {
+        const serviceInfo = {
+            '2#YpQjUw#NW_05_BLAC_2_1#05/04/2020': 'POULTON - BLACKPOOL via Victoria Hospital Outpatients',
+        };
+
+        const { req, res } = getMockRequestAndResponse({
+            cookieValues: {},
+            body: { ...serviceInfo },
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/tests/pages/api/serviceList.test.ts:144]: Do we need the spread (...) operator here?

File: repos/fdbt-site/src/pages/serviceList.tsx [hunk 4]
```diff
@@ -183,12 +189,36 @@ export const getServerSideProps = async (ctx: NextPageContextWithSession): Promi
         };
     });
 
+    const ticketJson = getSessionAttribute(ctx.req, MATCHING_JSON_ATTRIBUTE);
+    const matchingJsonMetaData = getSessionAttribute(ctx.req, MATCHING_JSON_META_DATA_ATTRIBUTE);
+
+    if (ticketJson && matchingJsonMetaData) {
+        const additional = false;
+        const multiOperator = false;
```
[the remark this answers is not in the document]
Comment (NitishAroraKPMG) [reply repos/fdbt-site/src/pages/serviceList.tsx]: made changes

**Review by RobCattonKPMG — APPROVED** [review]

File: repos/fdbt-site/tests/pages/serviceList.test.tsx [hunk 5]
```diff
@@ -161,6 +164,27 @@ describe('pages', () => {
                         hasBods: true,
                     }}
                     additional={false}
+                    editMode={false}
+                />,
+            );
+            expect(tree).toMatchSnapshot();
+        });
+
+        it.only('should render correctly when in edit mode', () => {
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/tests/pages/serviceList.test.tsx:173]: .only

Comment (NitishAroraKPMG) [reply repos/fdbt-site/tests/pages/serviceList.test.tsx]: removed and re-pushed

**Review by RobCattonKPMG — APPROVED** [review]

**Pull request merged**